### ISSUE TITLE
Port global-workflow to Jet

### ISSUE THREAD

**Issue by KateFriedman-NOAA** [NOAA-EMC/global-workflow#357]
This issue will document efforts to port global-workflow to Jet. @DavidHuber-NOAA will be working on porting to Jet.

Previously work done by @lgannoaa to port free-forecast mode was successful. @lgannoaa please send your Jet port changes back to the respective repos, including global-workflow. @DavidHuber-NOAA will use that to start with. Thanks!

Will need to establish glopara installs like other supported machines (@KateFriedman-NOAA can assist with):

- [x] GDA
- [x] obsproc packages
- [x] TC_tracker package (issue #1463 will take care of this item)

**Comment by DavidHuber-NOAA** [comment]
@lgannoaa The gldas build modules all look correct to me.  I did a test build with them and everything built fine.

By the way, I have not been able to start this work on Jet just yet.  I'm finishing up the S4 port, but I should be able to move to this in 2-3 weeks.

**Comment by lgannoaa** [comment]
@DavidHuber-NOAA 
Thank you for verify.

**Comment by DavidHuber-NOAA** [comment]
@lgannoaa I may have spoken too soon.  On S4 (#138), I was having issues with the gdasgldas job crashing.  The fix there was to update the esmf library to 8_1_1 in both gdas2gldas.s4 and gldas2gdas.s4.  The reason being that module_base.s4 is sourced, which loads esmf/8_1_1.  However, I notice that this mismatch is present in the gdas2gldas and gldas2gdas build scripts for all machines.  @KateFriedman-NOAA Should these be updated for all machines?

**Comment by KateFriedman-NOAA** [comment]
@DavidHuber-NOAA What tag of GLDAS are you using? The current tag of GLDAS in the develop branch ([gldas_gfsv16_release.v1.15.0](https://github.com/NOAA-EMC/GLDAS/releases/tag/gldas_gfsv16_release.v1.15.0)) uses `esmf/8_1_0_beta_snapshot_27` from hpc-stack. For Jet...the GLDAS will just need new modulesfiles that point to the same libraries through the hpc-stack installation there. In theory the new modulefile for gdas2gldas (gdas2gldas.jet) would be identical to the others but with the Jet hpc-stack installation path:

`module use /lfs4/HFIP/hfv3gfs/nwprod/hpc-stack/libs/modulefiles/stack`

(Note: the official hpc-stack installation locations are listed here: https://github.com/NOAA-EMC/hpc-stack/wiki/Official-Installations)

You should be able to just copy the hera one, change the module use line, and it will build (this work would happen in a branch in the GLDAS repo). The Jet part of sorc/machine-setup.sh and the module_base.jet in global-workflow will need to be updated/created too. Here is the gdas2gldas.hera modulefile from that tag as an example:

https://github.com/NOAA-EMC/GLDAS/blob/850af240db234b1bc8894e85650164d15ca344af/modulefiles/gdas2gldas.hera

Let me know if I didn't answer your question.

**Comment by DavidHuber-NOAA** [comment]
@KateFriedman-NOAA I am using the gldas_gfsv16_release.v1.15.0 tag.  You're correct that the module files in gldas all reference `esmf/8_1_0_beta_snapshot_27`.  What I was asking is whether this should be incremented to `esmf/8_1_1` since the workflow job gdasgldas (i.e. [gldas.sh](https://github.com/NOAA-EMC/global-workflow/blob/8c777ad04bce0555bed9f6666b808515b0a3766d/jobs/rocoto/gldas.sh#L5)) calls [load_fv3gfs_modules.sh](https://github.com/NOAA-EMC/global-workflow/blob/8c777ad04bce0555bed9f6666b808515b0a3766d/ush/load_fv3gfs_modules.sh#L20) which loads [module_base.machine](https://github.com/NOAA-EMC/global-workflow/blob/8c777ad04bce0555bed9f6666b808515b0a3766d/modulefiles/module_base.hera#L29) that loads the `esmf/8_1_1` module.

On S4, this setup was causing the gdas2gldas executable to crash in ESMF.  Incrementing the build scripts to `esmf/8_1_1` fixed the crash.

**Comment by KateFriedman-NOAA** [comment]
@DavidHuber-NOAA Oh, I see what you're saying now. Hmmmm, all of the components in develop have a mix of `esmf/8_1_0_beta_snapshot_27` and `esmf/8_1_1`. I can't make the decisions for the GLDAS folks but if `esmf/8_1_1` works on S4 then it should likely work elsewhere. Ok yeah, when you make S4/Jet changes in a GLDAS branch change the esmf module to `esmf/8_1_1` and then it can be discussed in the PR review.

We will be doing an overhaul of modules in workflow in the near future so we may push all components to `esmf/8_1_1`, I can't say for sure yet though since it'll be up to the component managers. The idea will be to move away from workflow-level modulefiles and rely on the component settings so this *may* become a moot point in the not too-distance future.

**Comment by DavidHuber-NOAA** [comment]
@KateFriedman-NOAA Alright, sounds like a plan.  Thanks!

**Comment by GeorgeGayno-NOAA** [comment]
@KateFriedman-NOAA I opened an issue for porting the gdas_init scripts to Jet. I have done no work on it yet. Should I prioritize it now?

https://github.com/NOAA-EMC/UFS_UTILS/issues/544

**Comment by KateFriedman-NOAA** [comment]
@GeorgeGayno-NOAA Yes, please now prioritize that task and keep @DavidHuber-NOAA up-to-date so he can test it when ready. UFS_UTILS is already working well enough to run the init job via global-workflow on Jet so we'll just need the gdas_init offline feature. Thanks!

**Comment by KateFriedman-NOAA** [comment]
@DavidHuber-NOAA I have installed the new obsproc packages on Jet that are going into operations next week (Issue #341). The updates for the new versions (HOMEobsproc* variables in config.base) will be going into the develop branch shortly after the implementation. I installed them in the existing glopara account space so, as long as BASE_GIT is correct for Jet, you shouldn't have to do anything for obsproc on Jet (except test the prep jobs). For Jet: `BASE_GIT=/lfs4/HFIP/hfv3gfs/glopara/git`

FYI, here are the installs:
/lfs4/HFIP/hfv3gfs/glopara/git/obsproc/obsproc_global.v3.4.2_hpc-stack
/lfs4/HFIP/hfv3gfs/glopara/git/obsproc/obsproc_prep.v5.5.0_hpc-stack

Let me know if you have any questions or issues whenever you get to testing the prep jobs in cycled mode via global-workflow on Jet. Thanks!

**Comment by DavidHuber-NOAA** [comment]
Will do, thanks @KateFriedman-NOAA!

**Comment by DavidHuber-NOAA** [comment]
@KateFriedman-NOAA Is there a rstprod directory for prepbufr data on Jet (i.e. GESROOT)?

**Comment by KateFriedman-NOAA** [comment]
> @KateFriedman-NOAA Is there a rstprod directory for prepbufr data on Jet (i.e. GESROOT)?

@DavidHuber-NOAA For GESROOT on Jet please set it to null like we're doing on Orion: `export GESROOT=/dev/null`

The `GESROOT` variable is specific mainly to WCOSS (it comes from the `prod_envir` module) but is also supported on Hera because NCO has a mirror of some production data from /com being pushed there. Now that we're expanding out to more machines we should try to make [this section of config.prepbufr](https://github.com/NOAA-EMC/global-workflow/blob/develop/parm/config/config.prepbufr#L13:L16) more generic.

Let's try something like this:
```
export GESROOT=${GESROOT:"/dev/null"}
if [ $machine = "HERA" ]; then
    export GESROOT="/scratch1/NCEPDEV/rstprod"
fi
```
When run on WCOSS after the `prod_envir` module is loaded the first line should get the path from the module. On Hera it should get that rstprod path and on other platforms it should be set to that null path since /com or a /com mirror isn't available elsewhere currently.

When your Jet work goes into PR I can run your branch on WCOSS to double check this adjustment doesn't mess anything up there. We'll want a similar test on Hera for the same reason. Thanks!

**Comment by DavidHuber-NOAA** [comment]
I have a working test build on Jet that has successfully run through 5 GDAS and 2 GFS cycles successfully running at C192/C96/L127 with 20 ensemble members.  The only errors I've received are related to the TC_Tracker software and some HPSS errors related to Jet's network problems yesterday and today.  The jobs should run over the weekend, at which point I can see how the gdasgldas job ran.

**Comment by DavidHuber-NOAA** [comment]
@KateFriedman-NOAA So far, I have only been running on xjet, but I have written up the resources for the other partitions.  Currently, I'm disabling C768 runs on ujet and vjet as the gfsfcst job is too large to run on ujet and would take up all of vjet.  The analysis jobs are also problematic for these partitions.  I could adjust threading and/or layouts in config.fv3 to allow C768 forecasts, but that would also probably require that the forecast hours be reduced (on S4, we can only run out to 168 hours @C384 with 1 thread).

Do you think I should keep C768 disabled on ujet and vjet or try to play around with the threading and/or layouts?

**Comment by KateFriedman-NOAA** [comment]
@DavidHuber-NOAA Thanks for this update! So since the login nodes build for `xjet` and `kjet` by default (what most folks will do) I'd like to make those two partitions our main focus for this task. As you've noted, the other partitions are too small for our largest jobs at the higher resolutions; they are all smaller than Hera, which is our smallest supported HPC right now with 24 cores/node.

Having said that though, I'd like to document the resource settings you've compiled and we can have a separate issue to extend support to other partitions after committing the cycled Jet support into develop. Question...how different are the resources for the other partitions? Is it more than just changing the max node value and letting the inner calculations happen...or are there additional adjustments? I'm trying to gauge how complicated the configs/resource settings will be to support the smaller partitions.

Thanks!

**Comment by DavidHuber-NOAA** [comment]
@KateFriedman-NOAA OK, that's easy enough.  I will adjust the config scripts for just those two, though I will note that @lgannoaa added support for `vjet` and `sjet` for the free-forecast.  I will leave that alone and add support for the cycled to just the `xjet` and `kjet` partitions.

For all of the partitions except `kjet`, some adjustments to resources are necessary for the `gdaseobs` and `(gdas|gfs)anal` jobs -- more importantly `eobs`.  At least one script requires the number of available cores to match the number of requested cores when creating/linking directories: [exglobal_atmos_analysis.sh](https://github.com/NOAA-EMC/GSI/blob/master/scripts/exglobal_atmos_analysis.sh#L614).  For instance, if gdaseobs is called at `C384` on `xjet` (default `npe=100`, `nth=2`), then it will be run on 9 nodes, or 216 cores.  This will result in an instability as some of the data generated by `gdaseobs` will not be copied back into the `ROTDIR`.  So the number of requested cores must match `npe`.  For the case above, `npe=108`.  @CoryMartin-NOAA may be able to elaborate if necessary.

Besides the resources, it may also be necessary to modify some of the build scripts to have the correct instructions for each architecture.  The Jet documentation lists these as `-axSSE4.2,AVX,CORE-AVX2,CORE-AVX512  -align array64byte -ip`.

**Comment by KateFriedman-NOAA** [comment]
> added support for vjet and sjet for the free-forecast. I will leave that alone and add support for the cycled to just the xjet and kjet partitions.

@DavidHuber-NOAA Great, thanks! Thanks also for the info on resource adjustments for some jobs on the other partitions and for the build instruction adjustments! I'll discuss this with my fellow workflow code managers and see how we'd like to proceed with supporting the other partitions beyond xjet and kjet.

**Comment by DavidHuber-NOAA** [comment]
@KateFriedman-NOAA Another job that will need modified resources for different partitions is gdaseupd.  So far, xjet is able to run the job without modification, though I've only tested it at C192/C96.  For kjet, I had to increase threads to 4 to run the C384/C192 as sacct showed a MAXRSS of ~8GB with 20 ensemble members.  Do you know the approximate memory footprint of this job at various resolutions?  If so, I can calculate what the threads will need to be for each partition.  Otherwise, I will just test and adjust as necessary.  Thanks!

**Comment by KateFriedman-NOAA** [comment]
@CoryMartin-NOAA @RussTreadon-NOAA What are the memory needs of the eupd job/execs? Dave is testing the cycled system on the kjet partition on Jet now after testing the xjet partition. Thanks!

kjet has 40ppn and 96 GB memory/node (similar to Hera). Here is the partition info for Jet:
https://jetdocs.rdhpcs.noaa.gov/wiki/index.php/Jet_System_Overview

**Comment by DavidHuber-NOAA** [comment]
@KateFriedman-NOAA @CoryMartin-NOAA @RussTreadon-NOAA Another job that is hitting memory limits on `xjet` is `gdasanalcalc`, in particular `interp_inc.x`, when running 10 tasks/node @768/384.  I have a solution for `xjet` in [calcanl_gfs.py](https://github.com/DavidHuber-NOAA/GSI/blob/ffd17841861bce94307314115c3bec1299fa14b3/ush/calcanl_gfs.py#L232) that spreads each instance of `interp_inc.x` across 2 nodes.  If we end up porting to other partitions, this can be modified to spread across more nodes, if necessary.  I suspect 768/384 will only be run on `xjet` and `kjet`, but it would be good to know what the memory requirements are for 384/192 since some of the older partitions have less memory per node.

**Comment by DavidHuber-NOAA** [comment]
I've successfully run a 384/192 on `kjet` for 5 cycles, including 2 GFS cycles, without any issues.  I will be starting a longer run today on `xjet`.

That said, I have so far only been running standard options.  Is there a desire to have support for `DO_BUFRSND`, `DO_GEMPAK`, `DO_AWIPS`, `WAFSF`, and/or any other features on Jet?

**Comment by KateFriedman-NOAA** [comment]
> I've successfully run a 384/192 on `kjet` for 5 cycles, including 2 GFS cycles, without any issues. I will be starting a longer run today on `xjet`.
> 
> That said, I have so far only been running standard options. Is there a desire to have support for `DO_BUFRSND`, `DO_GEMPAK`, `DO_AWIPS`, `WAFSF`, and/or any other features on Jet?

@DavidHuber-NOAA Thanks for the kjet update! So DO_BUFRSND, DO_GEMPAK, DO_AWIPS, and WAFSF are only required on the production machines (WCOSS) so you don't need to test them on Jet. We'll want to test the tracker though, let me get that installed and give you the location when ready. That's one of my checklist items on this issue.

**Comment by DavidHuber-NOAA** [comment]
Issues related to this port:

Port GLDAS: NOAA-EMC/GLDAS#25
Update the GSI and enable regression tests: NOAA-EMC/GSI#215
Port the UPP (completed): NOAA-EMC/UPP#380 & NOAA-EMC/UPP#381

**Comment by DavidHuber-NOAA** [comment]
During testing on kjet (and replicated on Hera), I found that the C48 efcs jobs required more memory to run in some cases.  At current resource allocations, some members will pass and some will fail with segmentation faults.  For kjet, which has the same RAM/CPU ratio as Hera of 2.4GB, I changed `nth_fv3` and `nth_fv3_gfs` to 2, which resolved the issue.  Should the default be changed for all systems or just Hera and Jet?

**Comment by KateFriedman-NOAA** [comment]
> I changed nth_fv3 and nth_fv3_gfs to 2, which resolved the issue. Should the default be changed for all systems or just Hera and Jet?

@DavidHuber-NOAA Yes, please change the default on all systems to use `nth_fv3[_gfs]=2` for C48 fcst. We don't test that resolution often so the change is likely needed everywhere. Thanks!

**Comment by DavidHuber-NOAA** [comment]
The C768 gfsfcst ran into some issues on Jet when performing parallel netCDF atm writes (sfc writes were performed serially for all tests).  The parallel writes would sometimes stall at irregular intervals, resulting in timed-out forecasts.  Rerunning the forecast for the same cycle generated all of the expected netCDF files, so it seems to be a transient, non-repeatable issue.  I observed this problem on xjet and @guoqing-noaa reported similar issues on kjet.  Users may expect to have to rerun the 768 if the jobs fail.

As a test, the C768 forecast was run a total of 4 times with serial netCDF writes on xjet.  These all completed successfully without any stalled netCDF writes.

**Comment by DavidHuber-NOAA** [comment]
The last three subcomponents have now been ported to Jet at the following commits:

GSI: [2f28fbf9cfc895a649b2517a25733e0e97a9d3d2](https://github.com/NOAA-EMC/GSI/commit/2f28fbf9cfc895a649b2517a25733e0e97a9d3d2)
GLDAS: [d1e437e4959639ddcdb069e6006f9fd77341d76d](https://github.com/NOAA-EMC/GLDAS/commit/d1e437e4959639ddcdb069e6006f9fd77341d76d)
UPP: [823343f73ff90ef5a40519fed034fffc4d5f1eb6](https://github.com/NOAA-EMC/UPP/commit/823343f73ff90ef5a40519fed034fffc4d5f1eb6)

**Comment by DavidHuber-NOAA** [comment]
The current authoritative develop branch as of September 9 has been ported to Jet and now resides on a fork ([port_2_jet](https://github.com/DavidHuber-NOAA/global-workflow/tree/port_2_jet)).  The following subcomponents have been tested on Jet at the respective hashes/tags:

- UFS (9350745855aebe0790813e0ed2ba5ad680e3f75c)
- GSI (9c1fc15d42573b398037319bbf8d5143ad126fb6) *
- GLDAS (gldas_gfsv16_release.v1.15.0) *
- UPP (ff42e0227d6100285d4179a2572b700fd5a959cb) *
- UFS_Utils (ufs_utils_1_6_0)
- Verif-global (verif_global_v2.5.2)

Three subcomponents have also been forked and ported with the ports residing in tags:

- GSI ([port_9c1fc15_2_jet](https://github.com/DavidHuber-NOAA/gsi/tree/port_9c1fc15_2_jet))
- UPP ([port_ff42e022_2_jet](https://github.com/DavidHuber-NOAA/UPP/releases/tag/port_ff42e022_2_jet))
- GLDAS ([port_v1.15.0_2_jet](https://github.com/DavidHuber-NOAA/gldas/tree/port_v1.15.0_2_jet))

Some portions of the global workflow have had an initial attempt at porting but have not been tested:
- WW3-related jobs
- postsnd
- getic

This fork has been tested at 96/48, 192/96, 384/192, and 768/384 resolutions, with extended tests performed at 192/96 and 384/192.

Caveats:
No attempt was made to port/test the GFS_WAFS repository, testing of VSDB was non-fruitful, and the TC Tracker software has not yet been installed on Jet.

**Comment by DavidHuber-NOAA** [comment]
@JiayiPeng-NOAA I have updated the Jet port of the TC_Tracker and I'm currently running a test with it.  Would you mind checking the install?  You can find it in /lfs1/NESDIS/nesdis-rdo2/David.Huber/TC_Tracker.

@jack-woollen I have ported the Fit2Obs package and I'm currently running a test with it.  Could you check the install for me?  You can find it in /lfs1/NESDIS/nesdis-rdo2/David.Huber/Fit2Obs.

**Comment by DavidHuber-NOAA** [comment]
@JiayiPeng-NOAA I finished a test run of the global workflow on Jet using the ported TC_Tracker.  Would you be able to look at the logs, products, and install?

Logs: /lfs1/NESDIS/nesdis-rdo2/David.Huber/para/com/test_f2o_tc2/logs
Archive products: /lfs1/NESDIS/nesdis-rdo2/David.Huber/archive/test_f2o_tc2
Install: /lfs1/NESDIS/nesdis-rdo2/David.Huber/TC_Tracker

Thanks!

**Comment by DavidHuber-NOAA** [comment]
@jack-woollen I finished a test run of the global workflow on Jet using the ported Fit2Obs package.  Would you be able to look at the logs, products, and install?

Logs: /lfs1/NESDIS/nesdis-rdo2/David.Huber/para/com/test_f2o_tc2/logs
Archive products: /lfs1/NESDIS/nesdis-rdo2/David.Huber/archive/test_f2o_tc2
Install: /lfs1/NESDIS/nesdis-rdo2/David.Huber/TC_Tracker

Thanks!

**Comment by jack-woollen** [comment]
David

I can't look on jet, no account. Hera, gaea, orion, niagara, wcoss, or
wcoss2 ok.

Jack

**Comment by JiayiPeng-NOAA** [comment]
Hi David,
I checked the GFS track/genesis data
files: /mnt/lfs1/NESDIS/nesdis-rdo2/David.Huber/archive/test_f2o_tc2/*atcf*
They look good to me.
Thanks,
Jiayi

**Comment by DavidHuber-NOAA** [comment]
@JiayiPeng-NOAA Thank you very much for checking!

@jack-woollen I have transferred the logs and the archive to Hera.  You can find them at /scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_logs and /scratch1/NESDIS/nesdis-rdo2/jet_archive.  Please let me know if you need any other data.  Thanks!

**Comment by jack-woollen** [comment]
David

As far as the fit2obs scripts and codes are concerned, they are running
correctly given the inputs. Ideally they would create forecast fit files
comparing time0 data with each of the forecasts valid at time0 from the 5
previous days, That is not happening probably because there are not 5
previous days in the forecast archive when each day0 fit is being run. That
should be simple to fix in the forecast archive cleanup scripts.

Jack

**Comment by DavidHuber-NOAA** [comment]
@jack-woollen Ah yes, I had set the forecasts to only run out 72 hours.  I'll modify that to 120 and rerun the cycling experiment.  Thanks!

**Comment by jack-woollen** [comment]
Forecasts need to run out to 126 hours because of time interpolation.  -Jack

**Comment by DavidHuber-NOAA** [comment]
OK, 126 hours then.  Thank you!

**Comment by DavidHuber-NOAA** [comment]
@jack-woollen I have completed the longer test.  The outputs and logs are in the same locations as before on Hera: /scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_logs and /scratch1/NESDIS/nesdis-rdo2/jet_archive

**Comment by jack-woollen** [comment]
David
The pattern of fit files valid at 2020080600 is correct for the case where
one verifying forecast is made every 24 hours at 00z for 5 previous days.
Jack

/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f00.raob.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f06.raob.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f24.raob.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f48.raob.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f72.raob.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f96.raob.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f120.raob.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f00.acft.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f06.acft.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f24.acft.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f48.acft.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f72.acft.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f96.acft.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f120.acft.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f00.acar.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f06.acar.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f24.acar.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f48.acar.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f72.acar.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f96.acar.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f120.acar.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f00.surf.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f06.surf.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f24.surf.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f48.surf.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f72.surf.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f96.surf.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f120.surf.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f00.sfc.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f06.sfc.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f24.sfc.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f48.sfc.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f72.sfc.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f96.sfc.2020080600
/scratch1/NESDIS/nesdis-rdo2/David.Huber/jet_archive/test_f2o_tc2/fits/f120.sfc.2020080600

**Comment by DavidHuber-NOAA** [comment]
Great, thank you for checking, @jack-woollen.

**Comment by DavidHuber-NOAA** [comment]
@JiayiPeng-NOAA Would it be possible to have the Jet port of the TC Tracker incorporated into the repository and a new tag made?

**Comment by JiayiPeng-NOAA** [comment]
Hi David,
You are welcome to create a new tracker tag in your own Github.
I don't know whether the HPC-stack was installed on JET or not.
The GFS workflow team did not send me a request to create a tag for JET.
Thanks,
Jiayi

**Comment by WalterKolczynski-NOAA** [comment]
> The GFS workflow team did not send me a request to create a tag for JET.

@JiayiPeng-NOAA I'll make one now then: please make a tag for Jet. hps-stack *is* on Jet, and should be what is used: https://github.com/NOAA-EMC/hpc-stack/wiki/Official-Installations

**Comment by JiayiPeng-NOAA** [comment]
Hi Walter,
Thanks for the update.
I usually work with Kate to create a new tracker tag. After I make a new
tracker version for JET, Kate needs to modify the GFSv16 workflow and test
the tracker jobs.
If everything is ok, a new TAG will be created.
Kate is busy with the WCOSS2 transition. I don't think she has time now.
After the GFSv16 WCOSS2 transition is done, I may start to work on this.
Thanks,
Jiayi

**Comment by DavidHuber-NOAA** [comment]
@KateFriedman-NOAA During the Jet testing, I had a failed call to global_gsi.x during a gdasanal job.  However, the job failed to exit and the gdasfcst was then called, which failed.  I've opened a GSI ticket: NOAA-EMC/GSI#293.

**Comment by KateFriedman-NOAA** [comment]
> @KateFriedman-NOAA During the Jet testing, I had a failed call to global_gsi.x during a gdasanal job. However, the job failed to exit and the gdasfcst was then called, which failed. I've opened a GSI ticket: [NOAA-EMC/GSI#293](https://github.com/NOAA-EMC/GSI/issues/293).

Roger that, thanks for the update @DavidHuber-NOAA !

**Comment by arunchawla-NOAA** [comment]
Hi Can someone provide a status on this ? @KateFriedman-NOAA @lgannoaa @WalterKolczynski-NOAA @DavidHuber-NOAA

**Comment by DavidHuber-NOAA** [comment]
@arunchawla-NOAA The global workflow code changes are complete.  I have completed cycled, ATM tests at C192/C96 (40-cycle) and C384/C192 (4-cycle) on xjet.  I still need to run a 1-cycle C768/C384 test on xjet and a 5-cycle C384/C192 kjet.

I have working ports of the TC_Tracker and Fit2Obs in my personal area on Jet that need to be incorporated into their respective authoritative repos.  Jiayi has requested to work with Kate on the TC_Tracker.  A request should be made for the Fit2Obs port as well.  They will then need to be installed in official locations.

I'm not currently focused on this port as I am finishing up the S4 port and waiting on the TC_Tracker and Fit2Obs packages.

**Comment by arunchawla-NOAA** [comment]
Thanks David, @JiayiPeng-NOAA and @jack-woollen can you work with @KateFriedman-NOAA and complete the updates to the authoritative repos for TC Tracker and Fit2Obs so we can close this task ?

**Comment by WalterKolczynski-NOAA** [comment]
What's the status on this?

**Comment by DavidHuber-NOAA** [comment]
@WalterKolczynski-NOAA I am wrapping up the revival of the S4 port presently.  After that, I will be able to come back and continue working on Jet.

**Comment by DavidHuber-NOAA** [comment]
@WalterKolczynski-NOAA @KateFriedman-NOAA I have restarted work on the Jet port.  Do you know if the TC tracker, Fit2Obs, prepobs, and obsproc have been installed on Jet?  If not, I can port, build, and test these locally.  I believe all of these now have GitHub repositories (though I'm not sure if Fit2Obs and the TC tracker are official), so I should be able to open PRs after testing.  I will just need them built in an official location at some point thereafter.

Does that sound like a reasonable plan?

**Comment by JiayiPeng-NOAA** [comment]
Hi David,
I will test TC tracker on JET. When it is finished I will create a TAG on
github for you to check out.
Thanks,
Jiayi

**Comment by DavidHuber-NOAA** [comment]
Great, thank you @JiayiPeng-NOAA!

**Comment by KateFriedman-NOAA** [comment]
@JiayiPeng-NOAA I see Jet support in the v1.1.15.5 tag. Is that not full support? I installed a copy on Jet here just now and it built ok (I think):

/lfs4/HFIP/hfv3gfs/glopara/git/TC_tracker/v1.1.15.5

Build log: /lfs4/HFIP/hfv3gfs/glopara/git/TC_tracker/v1.1.15.5/sorc/build.log

Happy to install a newer version when available, thanks!

@DavidHuber-NOAA related to tracker, I updated the syndat folder on Jet with the latest copy from Hera. You can set COMINsyn to this: /lfs4/HFIP/hfv3gfs/glopara/com/gfs/prod/syndat

In the `jet.yaml` hosts file:

```
comroot: '/lfs4/HFIP/hfv3gfs/glopara/com'
cominsyn: '${COMROOT}/gfs/prod/syndat'
```

For the other components:
* fit2obs - the newer version isn't on Jet yet, please work with Jack Woollen to add Jet updates into fit2obs repo
* obsproc - not yet on Jet, the v1.0.2 doesn't support Jet yet so please install it locally, make the needed changes to build it on Jet (e.g. create modulefile), here is the Hera copy: /scratch1/NCEPDEV/global/glopara/git/obsproc/v1.0.2
* prepobs - same as obsproc, doesn't yet support Jet and would need updates for Jet, here is the Hera copy: /scratch1/NCEPDEV/global/glopara/git/prepobs/v1.0.1

Once you have building/working versions of obsproc/prepobs on Jet we can work with the OBSPROC group to get those changes into their repos. Won't make it in time for the final obsproc.v1.1.0 tags to get cut but we can use locally modified copies on Jet for now and get the changes into their repos for future tags.

Once these external components are buildable and working on Jet I can install versions in the glopara space and the workflow can point to them there. 

Let me know what other external pieces you may need or questions about the contents of the `jet.yaml` hosts files you may need. Thanks!

**Comment by JiayiPeng-NOAA** [comment]
Hi Kate,
The version 1.1.15.5 should work for JET if there are no libraries updated
for JET recently.
As I worked with Lin.Gan for the GFSv16.3/ GFSv17 workflow, I will have
some updates for TC_tracker.v.1.1.15.5.
I will create a new version working for Wcoss2/Jet/Hera.
Thanks,
Jiayi

**Comment by KateFriedman-NOAA** [comment]
Ok thanks @JiayiPeng-NOAA ! Please include Orion in the next version as well, thanks!

**Comment by DavidHuber-NOAA** [comment]
Currently, so far as I know, there isn't a fully supported Python distribution (i.e. intelpython) on Jet with all of the modules required for the global workflow.  So I'm going to follow @WalterKolczynski-NOAA's solution for hera and build an hpc-stack version of miniconda with a custom environment.

**Comment by WalterKolczynski-NOAA** [comment]
@DavidHuber-NOAA The official stack installation should have miniconda. If not, contact NCEP-LIBS team (or maybe it is the EPIC team now). The reason I had to create the custom stack was related to MET/MET+.

**Comment by DavidHuber-NOAA** [comment]
@WalterKolczynski-NOAA Good to know, thanks.  Miniconda is available under the NCEP-LIBS hpc-stack, but it is pretty vanilla and does not include, for instance, the netCDF4 module, nor are there any additional environments installed.  The EPIC version seems to be broken (unless I'm looking in the wrong place):

```
module use /lfs4/HFIP/hfv3gfs/nwprod/hpc-stack.epic/libs/intel/hpc-stack/modulefiles/stack
module load hpc/1.2.0
module load hpc-miniconda3/4.6.14
Lmod has detected the following error:  Cannot load module "hpc-miniconda3/4.6.14" without these module(s) loaded:
   miniconda3/4.6.14

While processing the following module(s):
    Module fullname        Module Filename
    ---------------        ---------------
    hpc-miniconda3/4.6.14  /lfs4/HFIP/hfv3gfs/nwprod/hpc-stack.epic/libs/intel/hpc-stack/modulefiles/core/hpc-miniconda3/4.6.14.lua
```
The miniconda3 module file does not exist.

Since the NCEP-LIBS version is the only one with intel 18, I'll reach out to the NCEP-LIBS team and request a new environment.  I'll also inform the EPIC team about miniconda.  Thanks.

**Comment by WalterKolczynski-NOAA** [comment]
> @WalterKolczynski-NOAA Good to know, thanks. Miniconda is available under the NCEP-LIBS hpc-stack, but it is pretty vanilla and does not include, for instance, the netCDF4 module, nor are there any additional environments installed. The EPIC version seems to be broken (unless I'm looking in the wrong place):

Yeah, NetCDF and others should come in through the ufswm environment, so that really needs to be installed.

**Comment by DavidHuber-NOAA** [comment]
I'm still waiting on a few hpc-stack installs on Jet to proceed:

g2 version 3.4.5 noaa-emc/hpc-stack#498
ESMF version 8_0_1 noaa-emc/hpc-stack#499
PIO version 2.5.7 noaa-emc/hpc-stack#500
miniconda3 w/ ufswm environment noaa-emc/hpc-stack#502

**Comment by DavidHuber-NOAA** [comment]
The UFS at [c5badea](https://github.com/ufs-community/ufs-weather-model/commit/c5badeabfb8d9200fe2f03b4cdcd6317e96fa1d8) includes a fix that enables coupled forecasts on all Jet partitions, not just kjet.  This isn't critical to continue the port, but will be nice to have.

**Comment by KateFriedman-NOAA** [comment]
@DavidHuber-NOAA Checking on obsproc and prepobs installs on Jet. I gave you some details in a comment back in October (see above). I can help coordinate bringing Jet support back to those repos if you have working copies. The obsproc just updated to v1.1.2 so we'll want to get that version on Jet now. Let me know the status and how I can help. Thanks!

**Comment by DavidHuber-NOAA** [comment]
@KateFriedman-NOAA Apologies for not getting back to you earlier on this.  I had ported and tested version ~1.0.1~ 1.0.2 on Jet.  I will do the same for 1.1.2 and let you know when they are ready this week.  I have not gotten to the fit2obs component yet.  I recently closed out a cloud project and with the recent build of the EPIC-managed hpc-stack on Jet, I should be able to wrap this up fairly soon.

**Comment by DavidHuber-NOAA** [comment]
@KateFriedman-NOAA There is one conflict in obsproc, though it seems to affect Hera and Orion as well.  In jobs/JOBSPROC_GLOBAL_PREP, lines 165-169, the job copies prod_util executables from ush/.  However, for prod_util 1.2.2, these are installed in bin/.  This doesn't seem to be a blocker since you were able to test it on Orion, so I will continue with the Jet build/test and ignore the errors then submit an issue to obsproc.

**Comment by KateFriedman-NOAA** [comment]
> Apologies for not getting back to you earlier on this. I had ported and tested version 1.0.1 1.0.2 on Jet. I will do the same for 1.1.2 and let you know when they are ready this week.

No worries, thanks! I can take those changes back to the obsproc folks to get them integrated.

> There is one conflict in obsproc, though it seems to affect Hera and Orion as well. In jobs/JOBSPROC_GLOBAL_PREP, lines 165-169, the job copies prod_util executables from ush/. However, for prod_util 1.2.2, these are installed in bin/.

@DavidHuber-NOAA Yeah, this is because obsproc updated to use those execs under `ush` (a change in prod_util.v2+) but we don't yet have prod_util.v2+ outside of WCOSS2. I had been locally changing this in obsproc installs to avoid those errors but the errors don't affect anything (when I tested obsproc.v1.1.*) so I didn't bother with the latest installs. Keep hoping we'll get a newer prod_util on these machines.

> I have not gotten to the fit2obs component yet.

This reminds me that I need to get back to fit2obs on WCOSS2 and some updates to get it working with recent changes to global-workflow (need to convert fit2obs to bash throughout).

**Comment by DavidHuber-NOAA** [comment]
@KateFriedman-NOAA I've tested Obsproc v1.1.2 and Prepobs 1.0.1.  Both seem to output all the necessary files for the analysis, though there are a couple new messages in the logs I do not recognize:
`Current OBSPROC version is not set`
`GFS version used by this job is not set`

I don't see anywhere in the obsproc code uses the `obsproc_ver` or `gfs_ver` variables in the current version, so I just wanted to double check that these don't need to be set.  If so, then I think they are ready to go.  You can find them here:
`/lfs1/NESDIS/nesdis-rdo2/David.Huber/obsproc/v1.1.2`
`/lfs1/NESDIS/nesdis-rdo2/David.Huber/prepobs/v1.0.1`

**Comment by KateFriedman-NOAA** [comment]
> Both seem to output all the necessary files for the analysis, though there are a couple new messages in the logs I do not recognize:
> `Current OBSPROC version is not set`
> `GFS version used by this job is not set`

I get those messages in my prep logs too and it's ok. You see those messages because we haven't yet introduced version files in the workflow `develop` branch, like we did for the operational GFSv16 system (see the versions folder and files in the `dev/gfs.v16` branch). Once we do start setting those versions in the workflow (via versions files) you'll see this in the prep logs:
```
logs/2022010200/gdasprep.log:Current OBSPROC version is v1.1
logs/2022010200/gfsprep.log:Current OBSPROC version is v1.1
logs/2022010200/gdasprep.log:GFS version used by this job is v16.2.0
logs/2022010200/gfsprep.log:GFS version used by this job is v16.2.0
```

As long as the correct COM* paths get set and the output can be produced we don't have to set those versions for development. See what I explain below...

> I don't see anywhere in the obsproc code uses the obsproc_ver or gfs_ver variables in the current version, so I just wanted to double check that these don't need to be set.

The obsproc scripts use `gfs_ver` for the default COM* paths, see the usage here in `jobs/JOBSPROC_GLOBAL_PREP`:
```
-bash-4.2$ grep gfs_ver jobs/JOBSPROC_GLOBAL_PREP
echo "GFS version used by this job is ${gfs_ver:-not set}"
#export HOMEgfs=${HOMEgfs:-${PACKAGEROOT:?}/gfs.${gfs_ver:?"###FATAL ERROR \"gfs_ver\" is not set"}}
export HOMEgfs=${HOMEgfs:-$(compath.py gfs/${gfs_ver})}
  export COMINtcvital=${COMINtcvital:-$(compath.py ${envir}/${NET}/${gfs_ver}/${RUN}.${PDY})/${cyc}/${COMPONENT}}
  export COMINtcvital=${COMINtcvital:-$(compath.py ${envir}/${NET}/${gfs_ver}/${RUN}.${PDY})/${cyc}/${COMPONENT}}
export COMINgdas=${COMINgdas:-$(compath.py ${envir_getges}/gfs/${gfs_ver})/gdas.$PDY/${cyc}/${COMPONENT}}
export COMINgfs=${COMINgfs:-$(compath.py ${envir_getges}/gfs/${gfs_ver})/gfs.$PDY/${cyc}/${COMPONENT}}
```

When we run obsproc/prepobs in development we normally set those COM* variables ourselves so the defaults aren't used. Operations uses those defaults and set the version variables at the ecFlow level.

To be safe, look at your logs and ensure that `COMINtcvital`, `COMINgdas`, and `COMINgfs` are set and point to your `ROTDIR` folders. Here are those path variables in the test I did on Orion for the new obsproc.v1.1.2 package:
```
Orion-login-1[23] /work/noaa/stmp/kfriedma/comrot/obsproc112$ grep COMINgdas logs/2022010200/gdasprep.log
+ prep.sh[99]: export COMINgdas=/work/noaa/stmp/kfriedma/comrot/obsproc112/gdas.20220102/00/atmos
+ prep.sh[99]: COMINgdas=/work/noaa/stmp/kfriedma/comrot/obsproc112/gdas.20220102/00/atmos
+ 0s + export COMINgdas=/work/noaa/stmp/kfriedma/comrot/obsproc112/gdas.20220102/00/atmos
+ 0s + COMINgdas=/work/noaa/stmp/kfriedma/comrot/obsproc112/gdas.20220102/00/atmos
COMINgdas=/work/noaa/stmp/kfriedma/comrot/obsproc112/gdas.20220102/00/atmos

Orion-login-1[24] /work/noaa/stmp/kfriedma/comrot/obsproc112$ grep COMINgfs logs/2022010200/gdasprep.log
+ prep.sh[100]: export COMINgfs=/work/noaa/stmp/kfriedma/comrot/obsproc112/gfs.20220102/00/atmos
+ prep.sh[100]: COMINgfs=/work/noaa/stmp/kfriedma/comrot/obsproc112/gfs.20220102/00/atmos
+ 0s + export COMINgfs=/work/noaa/stmp/kfriedma/comrot/obsproc112/gfs.20220102/00/atmos
+ 0s + COMINgfs=/work/noaa/stmp/kfriedma/comrot/obsproc112/gfs.20220102/00/atmos
COMINgfs=/work/noaa/stmp/kfriedma/comrot/obsproc112/gfs.20220102/00/atmos

Orion-login-1[25] /work/noaa/stmp/kfriedma/comrot/obsproc112$ grep COMINtcvital logs/2022010200/gdasprep.log
++ config.prep[21]: export COMINtcvital=/work/noaa/rstprod/dump/gdas.20220102/00/atmos
++ config.prep[21]: COMINtcvital=/work/noaa/rstprod/dump/gdas.20220102/00/atmos
+ 0s + export COMINtcvital=/work/noaa/rstprod/dump/gdas.20220102/00/atmos
+ 0s + COMINtcvital=/work/noaa/rstprod/dump/gdas.20220102/00/atmos
COMINtcvital=/work/noaa/rstprod/dump/gdas.20220102/00/atmos
```

Note, that test reproduced the prepbufr file from a control using the older obsproc package, since I was running with a date before the newer obsproc package became relevant (new obs in ops with v16.3 last fall).

I will pull your installs under the glopara space, make sure I can run their builds successfully with your changes, and report back. Thanks!

**Comment by DavidHuber-NOAA** [comment]
OK, thanks for the tips.  The gfsprep and gdasprep logs both identify the ROTDIR correctly for COMINtcvital, COMINgdas, and COMINgfs:
```
+ 0s + export COMINtcvital=/lfs1/NESDIS/nesdis-rdo2/David.Huber/save/dump/gdas.20220401/00/atmos
+ 0s + COMINtcvital=/lfs1/NESDIS/nesdis-rdo2/David.Huber/save/dump/gdas.20220401/00/atmos
+ 0s + export COMINgdas=/lfs1/NESDIS/nesdis-rdo2/David.Huber/para/com/epic_96/gdas.20220401/00/atmos
+ 0s + COMINgdas=/lfs1/NESDIS/nesdis-rdo2/David.Huber/para/com/epic_96/gdas.20220401/00/atmos
+ 0s + export COMINgfs=/lfs1/NESDIS/nesdis-rdo2/David.Huber/para/com/epic_96/gfs.20220401/00/atmos
+ 0s + COMINgfs=/lfs1/NESDIS/nesdis-rdo2/David.Huber/para/com/epic_96/gfs.20220401/00/atmos
```

**Comment by KateFriedman-NOAA** [comment]
I pulled your obsproc and prepobs installs here under the glopara account space on Jet:
```
/lfs4/HFIP/hfv3gfs/glopara/git/obsproc/v1.1.2
/lfs4/HFIP/hfv3gfs/glopara/git/prepobs/v1.0.1
```

I was able to successfully build them. Thanks for making the needed changes to accomplish this!

I looked at the git diffs in both packages to see what would need to get sent back to the respective repos to support Jet:

- obsproc: 

Looks simple and sufficient, appreciate the local change for the `UTILROOT` ush->bin paths in `jobs/JOBSPROC_GLOBAL_PREP`. No questions from me on these changes. I will be sending back the `ush/build.sh` update and the new `modulefiles/obsproc_jet.lua` file. We'll need to maintain the local edit for ush->bin for now, until we get the newer prod_util on Jet.

- prepobs:

Updates to `README.md`, `build.sh`, and introduction of `modulefiles/prepobs_jet.lua` look good. I uncommented the two lines you had commented out in `build.sh` to clean out the `/build`  folder once the build was complete (to keep this install clean).

Questions: What is the change to the `fix/CMakeLists.txt` file? Was there an issue obtaining those fix files during install? I didn't encounter any issues during my other platform installs but I installed this version on the other platforms last summer (so not recently).
```
.../hfv3gfs/glopara/git/prepobs/v1.0.1> git diff fix/CMakeLists.txt
diff --git a/fix/CMakeLists.txt b/fix/CMakeLists.txt
index 2f0c98f..56f888a 100644
--- a/fix/CMakeLists.txt
+++ b/fix/CMakeLists.txt
@@ -1,6 +1,6 @@
 # Set URL, sha256sum and tar name of the file to download.
 set(URL "https://ftp.emc.ncep.noaa.gov/static_files/public/obsproc")
-set(SHA "1dd0bbaf7ab7452de9eb5949595540b20abc4e59439567a87c0b007de90fab3c")
+set(SHA "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855")
 string(SUBSTRING ${SHA} 0 6 SHORTSHA)
 set(TAR "obsproc-fix-${SHORTSHA}.tgz") # poor-man's version control
 
@@ -33,11 +33,8 @@ list(APPEND fix_files
   prepobs_prep.bufrtable
   satellite_counts.tbl
   syndat_weight
-  )
-
-list(APPEND fix_files
-  ${CMAKE_CURRENT_BINARY_DIR}/fix/prepobs_landc
-  ${CMAKE_CURRENT_BINARY_DIR}/fix/syndat_syndata.slmask.t126.gaussian
+  prepobs_landc
+  syndat_syndata.slmask.t126.gaussian
   )
```

Ignoring my question about the CMake file for this statement: I will be sending back the updates to `README.md` / `build.sh`, and new file `modulefiles/prepobs_jet.lua` to the prepobs repo.

**Comment by DavidHuber-NOAA** [comment]
@KateFriedman-NOAA I see the problem with the fix files.  I think I misinterpreted this error:
```
  file DOWNLOAD HASH mismatch

    for file: [/lfs1/NESDIS/nesdis-rdo2/David.Huber/prepobs/v1.0.1_cp/build/fix/obsproc-fix-1dd0bb.tgz]
      expected hash: [1dd0bbaf7ab7452de9eb5949595540b20abc4e59439567a87c0b007de90fab3c]
        actual hash: [e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855]
             status: [28;"Timeout was reached"]
```
to indicate that the hash did not exist on the FTP site.  But what it is actually telling me is that Jet cannot connect to the site.  So by changing the hash to e3b0c44..., I tricked cmake (and myself) into thinking that it successfully downloaded the fix files.  I wonder if there is a way to stage this data or skip this step completely then place it in the fix directory after the build.  Alternatively, we could get an exception on Jet for this FTP site.

**Comment by DavidHuber-NOAA** [comment]
I've gone ahead and made the request to Jet support.

**Comment by KateFriedman-NOAA** [comment]
@DavidHuber-NOAA Thanks for confirming there is an access issue and for reaching out to the Jet folks to get it sorted. Let me know how it goes!

I'll proceed with opening obsproc/prepobs issues and getting your Jet-support changes into branches in each repo. I gave the OBSPROC team a heads-up about this coming in earlier this week so it won't be a surprise to them. They said their next operational hand-off (and therefore tag that will be cut) will be this spring so we won't have a new official version to install before then. We can use the installs you did and which I copied into glopara space until then. When the next tags are cut I will install them on Jet at the same time I install them everywhere else.

Other than obsproc stuff...when you have a moment, can you provide a fresh status update on where the port is and what is left to do. I have some more time now and can assist with other external package installs under glopara or GDA population as needed. I think I will treat the GDA on Jet similarly to how I treat the Orion GDA = populate it with time periods requested by users and not add huge chunks or real-time. Thanks!

**Comment by DavidHuber-NOAA** [comment]
@KateFriedman-NOAA Sure thing.  I just completed a 21 cycle C96/C48 test on Jet.  The gdasgldas job is failing due to an apparent linking problem in ESMF.  I've notified hpc-stack (noaa-emc/hpc-stack#499).  The verif-global jobs are also silently failing and that repo probably needs an update for Jet.  Fit2Obs did not run correctly -- I think I just need to load prod_util so it can find `postmsg` as the job stops abruptly when it tries to call it at line 46 of excfs_gdas_vrfyfits.sh.ecf.  The gfs post jobs show a non-failing octal error for forecast hours >= 9 and < 100 (reported in #1195).

So to summarize, we are successfully cycling with a few bumps, but I think I can solve them fairly easily.  I'll start up a C192/C96 test next when I have the fixes in place.

**Comment by KateFriedman-NOAA** [comment]
@DavidHuber-NOAA Great, thanks for that update! I need to send some fit2obs updates to that repo for WCOSS2 so we can include Jet updates in it too. I'll ping you in that repo as I work on it...waiting for the WCOSS2 maintenance to finish so I can start testing there again.

If you want to send initial changes back to global-workflow `develop` before it's 100% sorted that would be ok. Smaller PRs are usually better, so if you wanted to break the Jet changes into chunks that would be fine. We have known fit2obs and vrfy issues across multiple platforms anyway that have to get sorted.

**Comment by DavidHuber-NOAA** [comment]
That sounds reasonable.  I will wrap up what I have presently -- and a few caveats -- into a PR and throw it over the fence.  I will probably disable metp and gldas jobs until I get those sorted.

**Comment by DavidHuber-NOAA** [comment]
With the new hpc-stack build location on Jet, I updated the module files for most of the components.  I will submit PRs for each of these early next week.  @KateFriedman-NOAA Would you like for me to open a draft PR with these component PRs as prerequisites or modify the checkout script to point to my forks of the current develop hashes when checking out on Jet?

**Comment by KateFriedman-NOAA** [comment]
> Would you like for me to open a draft PR with these component PRs as prerequisites

Let's do this option. We're about to move to newer hashes for the ufs-weather-model and GSI as part of moving to the newer crtm/2.4.0 anyway (issue #1233). Is that crtm version on Jet?

**Comment by DavidHuber-NOAA** [comment]
@KateFriedman-NOAA No, that version isn't on Jet yet, but I will request it now and notify @ulmononian.

**Comment by KateFriedman-NOAA** [comment]
@DavidHuber-NOAA Ok, glad I asked, thanks! :)

**Comment by KateFriedman-NOAA** [comment]
@DavidHuber-NOAA For issue #1233 I plan to update all existing workflow module_base* modulefiles to load `crtm/2.4.0`. Will see how the updates align with regards to the version in the Jet modulefile.

**Comment by DavidHuber-NOAA** [comment]
@KateFriedman-NOAA Noted, thanks.  S4 should be good to go on that front, but I can give it a test run when ready.

**Comment by KateFriedman-NOAA** [comment]
@DavidHuber-NOAA The hpc-stack path in the obsproc/prepobs installs I got from you last week are using the new build location? I am going to open obsproc/prepobs issues and make branches for those changes today. Thanks!

**Comment by DavidHuber-NOAA** [comment]
@KateFriedman-NOAA Yes, they are both using the updated location of hpc-stack.

**Comment by JiayiPeng-NOAA** [comment]
Hi Kate,
Please tell me the new hps-stack build location on JET. I may need it to
update TC-tracker package.
Thanks,
Jiayi

**Comment by DavidHuber-NOAA** [comment]
@JiayiPeng-NOAA that is /lfs4/HFIP/hfv3gfs/role.epic/hpc-stack/libs/intel/18.0.5.274/modulefiles/stack

**Comment by KateFriedman-NOAA** [comment]
@DavidHuber-NOAA Excellent, thanks for confirming and for sharing the path in here for others.

**Comment by JiayiPeng-NOAA** [comment]
Hi David,
Thanks for the information!
Jiayi

**Comment by JiayiPeng-NOAA** [comment]
Hi Kate,
I also need to know the path of the New HPC-stack on Hera, Orion.
Do we have HPC-stack on wcoss2?
Thanks for the help!
Jiayi

**Comment by KateFriedman-NOAA** [comment]
@JiayiPeng-NOAA The hpc-stack installs that we are currently using on the other platforms can be found at the top of the other module_base files in this folder of global-workflow develop: https://github.com/NOAA-EMC/global-workflow/tree/develop/modulefiles

Note, the WCOSS2 production libraries are the stack there.

We haven't moved to any newer EPIC-installed ones (like the one on Jet) on those platforms yet nor have we move to any spack-stack installs yet...although that is coming in the near future. When we do update to other stack installs we'll open global-workflow issues so keep an eye for that this year.

**Comment by KateFriedman-NOAA** [comment]
Opened obsproc (https://github.com/NOAA-EMC/obsproc/issues/78) and prepobs (https://github.com/NOAA-EMC/prepobs/issues/15) issues to get Jet support into those repos.

**Comment by DavidHuber-NOAA** [comment]
@KateFriedman-NOAA Could you push the GSI fix data from Hera to Jet?  It would be nice to have for the upcoming PR since I'm already going to be changing the module file.

**Comment by KateFriedman-NOAA** [comment]
@DavidHuber-NOAA What GSI fix data exactly (path on Hera)? The full fix set (with gdas and gsi subfolders) is on Jet already:

```
/lfs4/HFIP/hfv3gfs/glopara/git/fv3gfs/fix/gdas
/lfs4/HFIP/hfv3gfs/glopara/git/fv3gfs/fix/gsi
```

Am I having a brain fart and forgetting more data that we stage for GSI on Hera? :)

**Comment by DavidHuber-NOAA** [comment]
@KateFriedman-NOAA Ah, thanks, I was just looking in the wrong place.

**Comment by JiayiPeng-NOAA** [comment]
Hi David and Kate,
I just created a TAG:
https://github.com/NOAA-EMC/TC_tracker/releases/tag/ens_tracker.v1.1.15.6
This package applies EPIC hpc-stack for compile and run on Hera/Jet/Orion.
Please update your GFS workflow.
If you have any questions, please let me know.
Thanks,
Jiayi

**Comment by KateFriedman-NOAA** [comment]
@JiayiPeng-NOAA Thanks for cutting this tag! I have installed it on Hera, Orion, both WCOSS2, and Jet.

Note, the Jet build failed right away with unknown platform. The `/lfs3` directory check did not work for me. I changed it to `/mnt/lfs1` (what we use in global-workflow) and it worked. Can you fix this in `sorc/build.sh` and recut the same tag name/number? Thanks!

**Comment by JiayiPeng-NOAA** [comment]
Hi Kate,
Fixed and created  with the same TAG name.
Thanks,
Jiayi

**Comment by KateFriedman-NOAA** [comment]
> Fixed and created  with the same TAG name.

Excellent, thanks @JiayiPeng-NOAA for the quick fix and recut tag! It builds out-of-the-box on Jet for me now. :)
